src/Peregrine/Peregrine.js

```diff
@@ -1,82 +1,36 @@
 import { createElement } from 'react';
-import { Provider } from 'react-redux';
-import { render } from 'react-dom';
+import { Provider as ReduxProvider } from 'react-redux';
 import createStore from '../store';
 import MagentoRouter from '../Router';
 
 /**
- * @class
- * @prop {Element} container
- * @prop {ReactElement} element
- * @prop {Store} store
- * @prop {string} apiBase
- * @prop {string} __tmp_webpack_public_path__
+ *
+ * @param {string} apiBase Absolute URL pointing to the GraphQL endpoint
+ * @param {string} __tmp_webpack_public_path__ Temporary hack. Expects the `__webpack_public_path__` value
+ * @returns {{ store: Store, Provider: () => JSX.Element }}
  */
-class Peregrine {
-    /**
-     * Create a Peregrine instance.
-     * @param {object} opts
-     * @param {string} opts.apiBase Base URL for the store's GraphQL API, including host/port
-     * @param {string} opts.__tmp_webpack_public_path__ Temporary until PWA module extends GraphQL API
-     */
-    constructor(opts = {}) {
-        const { apiBase = location.origin, __tmp_webpack_public_path__ } = opts;
-        this.apiBase = apiBase;
-        this.__tmp_webpack_public_path__ =
-            __tmp_webpack_public_path__ &&
-            ensureDirURI(__tmp_webpack_public_path__);
-        if (!__tmp_webpack_public_path__ && process.env.NODE_ENV === 'test') {
-            // Since __tmp_webpack_public_path__ is temporary, we're
-            // defaulting it here in tests to lessen tests that need to change
-            // when this property is removed
-            this.__tmp_webpack_public_path__ = 'https://temporary.com/pub/';
-        }
-        this.store = createStore();
-        this.container = null;
-        this.element = null;
-    }
-
-    /**
-     * Create an instance of the root component, wrapped with store and routing
-     * components.
-     *
-     * @returns {ReactElement}
-     */
-    render() {
-        const { store, apiBase, __tmp_webpack_public_path__ } = this;
-
-        return (
-            <Provider store={store}>
-                <MagentoRouter {...{ apiBase, __tmp_webpack_public_path__ }} />
-            </Provider>
+export default function bootstrap({ apiBase, __tmp_webpack_public_path__ }) {
+    // Remove deprecation warning after 2 version bumps
+    if (process.env.NODE_ENV !== 'production' && this instanceof bootstrap) {
```

Comment:
What's the scenario in which is this error thrown? I see the intended usage for the new API, but I don't understand the deprecated usage here.

Reply:
It's to catch people that have a theme currently doing `new Peregrine` with the default export. That `instanceof` check catches it, so we can let them know that their code needs to be changed.